test/test_jit.py

```diff
@@ -7344,6 +7313,87 @@ def func(x):
 ])
 
 
+def add_autograd_test(
+    postfix,
+    fn_creator,
+    run_magic_methods,
+    exclude_list,
+    name,
+    self_size,
+    new_args,
+    test_name,
+    output_process_fn,
+    skipTestIf,
+    kwargs,
+):
+    def do_test(
+        self,
+        name=name,
+        self_size=self_size,
+        args=new_args,
+        test_name=test_name,
+        output_process_fn=output_process_fn,
+    ):
+        def check(name):
+            is_magic_method = name[:2] == "__" and name[-2:] == "__"
+            is_inplace = name[-1] == "_" and not is_magic_method
+            self_variable = create_input((self_size,))[0][0]
+            # FixMe: run grad checks on inplace self
+            if is_inplace:
+                self_variable.requires_grad = False
+            # need to record this because methods can change the size (e.g. unsqueeze)
+            args_variable, kwargs_variable = create_input(args, requires_grad=not is_inplace, call_kwargs=kwargs)
+            self_tensor = deepcopy(self_variable.data)
+            args_tensor = deepcopy(unpack_variables(args_variable))
+            output_variable = getattr(self_variable, name)(*args_variable, **kwargs_variable)
+
+            def fn(*inputs, **kwargs):
+                output = getattr(inputs[0], name)(*inputs[1:], **kwargs)
+                return output_process_fn(output)
+
+            if not is_inplace and name not in EXCLUDE_GRADCHECK and not exclude_tensor_method(name, test_name):
+                if run_magic_methods or not is_magic_method:
+                    check_against_reference(
```

Comment:
This is change a copy-paste of the existing `do_test` logic, with it only running either a test with the tracer or as a script, breaking the autograd tests into 2 jit tests (with `_script` or `_traced` after the test name).